lib/src/main/java/com/cops/iitbhu/previewer/lib/Previewer.kt

```diff
@@ -115,4 +115,40 @@ object Previewer {
             .into(imageView)
     }
 
+    /**
+     * Generates a bitmap for remote video file
+     * @param uri Uri of the remote video file
+     * @return Bitmap of the first frame of the video corresponding to given uri
+     */
+    suspend fun getThumbnailFromRemoteVideoUri(uri: Uri): Bitmap? {
+        return withContext(Dispatchers.IO) {
+            try {
+                val mMR = MediaMetadataRetriever()
+                mMR.setDataSource(uri.toString(), mapOf())
+                val bitmap = mMR.frameAtTime
+                bitmap
+            } catch (e: Exception) {
+                null
+            }
+        }
+    }
+
+    /**
+     * Generates a bitmap for local video file
+     * @param uri Uri of the local video file
+     * @return Bitmap of the first frame of the video corresponding to given uri
+     */
+    suspend fun getThumbnailFromLocalVideoUri(uri: Uri): Bitmap? {
+        return withContext(Dispatchers.IO) {
```

Comment:
I guess we don't need to go to a background thread while working with local uri. Make it a normal function instead of a suspend one. Check if this works or not.
```kotlin
fun getThumbnailFromLocalVideoUri(uri: Uri): Bitmap? {
    try {
        val mMR = MediaMetadataRetriever()
        mMR.setDataSource(appContext, uri)
        mMR.frameAtTime
    } catch (e: Exception) {
        null
    }
}
```